src/Microsoft.IdentityModel.Tokens/TokenHandler.cs

```diff
@@ -125,6 +124,18 @@ internal virtual ClaimsIdentity CreateClaimsIdentityInternal(SecurityToken secur
                         MarkAsNonPII("internal virtual ClaimsIdentity CreateClaimsIdentityInternal(SecurityToken securityToken, TokenValidationParameters tokenValidationParameters, string issuer)"),
                         MarkAsNonPII(GetType().FullName))));
         }
+
+        internal static ClaimsIdentity CreateCaseSensitiveClaimsIdentityFromTokenValidationParameters(SecurityToken securityToken, TokenValidationParameters validationParameters, string issuer)
+        {
+            ClaimsIdentity identity = validationParameters.CreateClaimsIdentity(securityToken, issuer);
```

Comment:
Why do you want to attempt to create from `validationParameters` and possibly end up creating a case-insensitive version? Just side effects?

Reply:
CreateClaimsIdentity can be overriden in derived TVP by devs (extra claims could be added, for ex.) We're honoring that here but also changing the type to CsClaimsIdentity if needed.